canonical-web-and-design/maas-ui
# feat(ui): create common EditableSection component

## Done

- Updated `TitledSection` component to be able to have the title in a sidebar
- Built `EditableSection` component, that handles showing an edit button in a titled section

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the Configuration tab of a machine
- Check that the config sections haven't changed (i.e. title is in sidebar, clicking Edit hides the button)
- Go to a fabric's details page and check that the TitledSections there have not changed either

## Fixes

Fixes #3779
